src/commands.rs

```diff
@@ -874,7 +890,9 @@ mod test {
                             "--change_descriptor", "wpkh(tpubDEnoLuPdBep9bzw5LoGYpsxUQYheRQ9gcgrJhJEcdKFB9cWQRyYmkCyRoTqeD4tJYiVVgt6A3rN6rWn9RYhR9sBsGxji29LYWHuKKbdb1ev/1/*)",
                             "create_tx", "--to", "n2Z3YNXtceeJhFkTknVaNjT1mnCGWesykJ:123456","mjDZ34icH4V2k9GmC8niCrhzVuR3z8Mgkf:78910",
                             "--utxos","87345e46bfd702d24d54890cc094d08a005f773b27c8f965dfe0eb1e23eef88e:1",
-                            "--utxos","87345e46bfd702d24d54890cc094d08a005f773b27c8f965dfe0eb1e23eef88e:2"];
+                            "--utxos","87345e46bfd702d24d54890cc094d08a005f773b27c8f965dfe0eb1e23eef88e:2",
+                            "--add_string","Hello BDK",
+                            "--add_data","SGVsbG8gQkRLCg==" ];
```

Comment:
The test cannot work like this because you have defined `aad_data` and `add_string` mutually exclusive..  So structopt is correctly giving this error
`The argument '--add_string <ADD_STRING>' cannot be used with '--add_data <ADD_DATA>'`

Instead you need to make separate tests.. But I think the correct way here is just allow both types..

Reply:
> Thanks for the quick update.. Unfortunately the test fails because you can't pass both options in the same command.. I have few more conceptual update requests regarding that in code comments..
> 
> Also I would like to ask for a bit more work here, because we are currently going through a lot of refactoring in the crate. And many of those are scattered around other open PRs. So adding this in master right now will create a lot of conflicts with other open PRs and have to be manually resolved again..
> 
> So instead, can you rebase this on top of #104?? You might have to resolve few conflicts, but you will be able to work on the final version of the crate after all refatorings..
> 
> Rebasing on #104 will also allow you to add the functional test for opreturn output creation with bdk-cli and we can then fix the full behavior there. look in the folder `tests/intergation.rs` after you rebase, to see a example of basic test sample..


-  Hi Raj, I'm rebasing #105 on top of #104 like you mentioned. Seems like the best option considering what you've the refactoring etc...
- Also, I'm reviewing #104 and test it on my local. 
- Thank you for the update.